mpt_trie/src/special_query.rs

```diff
@@ -101,16 +99,16 @@ fn pop_nibbles_clamped(nibbles: &mut Nibbles, n: usize) -> Nibbles {
 /// Note that if the key does not match the entire key of a node (eg. the
 /// remaining key is `0x34` but the next key is a leaf with the key `0x3456`),
 /// then the leaf will not appear in the query output.
-pub fn path_for_query<K, T: PartialTrie>(
-    trie: &Node<T>,
+pub fn path_for_query<K>(
+    trie: &Node,
     k: K,
     always_include_final_node_if_possible: bool,
-) -> TriePathIter<T>
+) -> TriePathIter
 where
     K: Into<Nibbles>,
 {
     TriePathIter {
-        curr_node: trie.clone().into(),
+        curr_node: Arc::new(trie.clone()),
```

Comment:
Non-blocking: Could we skip cloning here and work with references?

Reply:
This is a bit late, but yeah tries in `mpt_trie` are all essentially references, so cloning is very cheap.